Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2797,10 +2789,16 @@ - (void)yapDatabaseModifiedExternally:(NSNotification *)notification
 
     DDLogVerbose(@"%@ %s", self.logTag, __PRETTY_FUNCTION__);
 
-    // External database modifications can't be converted into incremental updates,
-    // so rebuild everything.  This is expensive and usually isn't necessary, but
-    // there's no alternative.
-    [self resetMappings];
+    if (self.shouldObserveDBModifications) {
```

Comment:
We only need to reset our view state if we observe `yapDatabaseModifiedExternally` while we are observing db changes.